crates/ln-dlc-node/src/node/payment_persister.rs

```diff
@@ -0,0 +1,92 @@
+use crate::HTLCStatus;
+use crate::MillisatAmount;
+use crate::PaymentFlow;
+use crate::PaymentInfo;
+use anyhow::Result;
+use lightning::ln::PaymentHash;
+use lightning::ln::PaymentPreimage;
+use lightning::ln::PaymentSecret;
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::sync::Mutex;
+use time::OffsetDateTime;
+
+pub trait PaymentPersister: Clone {
+    fn insert(&self, payment_hash: PaymentHash, info: PaymentInfo) -> Result<()>;
+    fn upsert(
+        &self,
+        payment_hash: &PaymentHash,
+        flow: PaymentFlow,
+        htlc_status: HTLCStatus,
+        preimage: Option<PaymentPreimage>,
+        secret: Option<PaymentSecret>,
+    ) -> Result<()>;
+    fn get(&self, payment_hash: &PaymentHash) -> Result<Option<(PaymentHash, PaymentInfo)>>;
```

Comment:
perhaps some comments would be good? e.g. why it can return an Result<Option<T>>